src/main/scala/swiss/dasch/api/ProjectsEndpointsHandler.scala

```diff
@@ -195,10 +195,18 @@ final case class ProjectsEndpointsHandler(
       )
 
   private val postBulkIngestUploadEndpoint: ZServerEndpoint[Any, ZioStreams] = projectEndpoints.postBulkIngestUpload
-    .serverLogic(principal => { case (shortcode, filenames, stream) =>
+    .serverLogic(principal => { case (shortcode, filename, stream) =>
       for {
         _ <- authorizationHandler.ensureAdminScope(principal)
-        _ <- bulkIngestService.uploadSingleFile(shortcode, filenames, stream).mapError { e =>
+        path <-
+          ZIO
+            .fromEither(PathOps.fromString(filename))
+            .map(_.normalize)
+            .filterOrFail(!_.toString.startsWith(".."))("Cannot traverse out of the upload directory")
```

Comment:
Wouldn't that allow traversing outside after traversing in, in case there _is_ a directory to traverse into? e.g. `foo/bar/../../../../sensitive`.

Reply:
The `_.normalize` before transforms your example path to `../../sensitive`.